bin/describe/describe_external_lib_deps.ml

```diff
@@ -0,0 +1,194 @@
+        Memo.parallel_map select.choices
+          ~f:(fun (choice : Dune_rules.Lib_dep.Select.Choice.t) ->
+            Memo.parallel_map
+              (Lib_name.Set.to_string_list choice.required
+              @ Lib_name.Set.to_string_list choice.forbidden)
+              ~f:(fun name ->
+                external_resolve db (Lib_name.of_string name) Kind.Optional)
+            >>| List.filter_opt)
+        >>| List.concat)
+  >>| List.concat
+
+let external_libs db dir libraries preprocess names package kind =
+  let open Memo.O in
+  let open Item in
+  let* lib_deps = external_lib_deps db libraries in
+  let+ lib_pps = external_lib_pps db preprocess in
+  Some { kind; dir; names; package; external_deps = lib_deps @ lib_pps }
+
+let libs db (context : Context.t) (build_system : Dune_rules.Main.build_system)
+    =
+  let { Dune_rules.Main.conf; contexts = _; _ } = build_system in
+  let open Memo.O in
+  let* dune_files =
+    Dune_rules.Dune_load.Dune_files.eval conf.dune_files ~context
+  in
+  Memo.parallel_map dune_files ~f:(fun (dune_file : Dune_rules.Dune_file.t) ->
+      Memo.parallel_map dune_file.stanzas ~f:(fun stanza ->
+          let dir = dune_file.dir in
+          match stanza with
+          | Dune_rules.Dune_file.Executables exes ->
+            external_libs db dir exes.buildable.libraries
+              exes.buildable.preprocess
+              (List.map exes.names ~f:snd)
+              exes.package Item.Kind.Executables
+          | Dune_rules.Dune_file.Library lib ->
+            external_libs db dir lib.buildable.libraries
+              lib.buildable.preprocess
+              [ Dune_rules.Dune_file.Library.best_name lib |> Lib_name.to_string
+              ]
+              (Dune_rules.Dune_file.Library.package lib)
+              Item.Kind.Library
+          | Dune_rules.Dune_file.Tests tests ->
+            external_libs db dir tests.exes.buildable.libraries
+              tests.exes.buildable.preprocess
+              (List.map tests.exes.names ~f:snd)
+              tests.exes.package Item.Kind.Tests
+          | _ -> Memo.return None)
+      >>| List.filter_opt)
+  >>| List.concat
+
+let external_resolved_libs setup super_context =
+  let open Memo.O in
+  let context = Super_context.context super_context in
+  let* scope = Dune_rules.Scope.DB.find_by_dir context.build_dir in
+  let db = Dune_rules.Scope.libs scope in
+  libs db context setup
+  >>| List.filter ~f:(fun (x : Item.t) -> not (x.external_deps = []))
+
+let to_dyn context_name external_resolved_libs =
+  let open Dyn in
+  Tuple [ String context_name; list Item.to_dyn external_resolved_libs ]
+
+let term =
+  let+ common = Common.term
+  and+ context_name = Common.context_arg ~doc:"Build context to use."
+  and+ _ =
+    Arg.(
+      value
+      & opt (some string) None
+      & info [ "lang" ] ~docv:"VERSION"
+          ~doc:
+            "This argument has no effect and is deprecated. It exists solely \
+             for backwards compatibility.")
```

Comment:
Might as well move this argument to `Describe_lang`. I see it duplicated in a few places.

Reply:
OK, I've moved it to `Describe_lang_compat`.